### PR TITLE
Follow-up on #1487

### PR DESCRIPTION
**Pull request by serban300**
Implementing the review comments for #1487 

Just one mention: For https://github.com/paritytech/parity-bridges-common/pull/1487#discussion_r922113182 the implementation didn't go as planned since we need to take the Weight as a generic param and then we can't define a const with that type. So I did some adjustments.

### CODE REVIEW

**Comment by svyatonik** [comment]
> Just one mention: For https://github.com/paritytech/parity-bridges-common/pull/1487#discussion_r922113182 the implementation didn't go as planned since we need to take the Weight as a generic param and then we can't define a const with that type. So I did some adjustments.

I'm not sure, I get the problem :) Can you elaborate, please. I've just tried that:
```rust
// in bridge_runtime_common::messages
pub trait TransactionEstimation<Weight> {
	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight>;
}

pub struct X<
	AccountId: MaxEncodedLen,
	const MaxConfirmationTxWeight: Weight,
	const ExtraStorageProofSize: u32,
	const TxExtraBytes: u32,
>(PhantomData<AccountId>);

impl<AccountId: MaxEncodedLen, const MaxConfirmationTxWeight: Weight, const ExtraStorageProofSize: u32, const TxExtraBytes: u32> TransactionEstimation<Weight> for X<AccountId, MaxConfirmationTxWeight, ExtraStorageProofSize, TxExtraBytes> {
	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight> {
		let inbound_data_size = InboundLaneData::<AccountId>::encoded_size_hint_u32(1, 1);
		MessageTransaction {
			dispatch_weight: MaxConfirmationTxWeight,
			size: inbound_data_size
				.saturating_add(ExtraStorageProofSize)
				.saturating_add(TxExtraBytes),
		}
	}
}

// in millau/runtime:
type MillauTransactionEstimation = bridge_runtime_common::messages::X::<
	bp_millau::AccountId,
	{ bp_millau::MAX_SINGLE_MESSAGE_DELIVERY_CONFIRMATION_TX_WEIGHT },
	{ bp_millau::EXTRA_STORAGE_PROOF_SIZE },
	{ bp_millau::TX_EXTRA_BYTES },
>;
```

And it compiles. TBH weight as a generic argument seems like an overengineering, but right now some weights update is coming, so let's not remove it now.

**Comment by serban300** [comment]
> > Just one mention: For [#1487 (comment)](https://github.com/paritytech/parity-bridges-common/pull/1487#discussion_r922113182) the implementation didn't go as planned since we need to take the Weight as a generic param and then we can't define a const with that type. So I did some adjustments.
> 
> I'm not sure, I get the problem :) Can you elaborate, please. I've just tried that:
> 
> ```rust
> // in bridge_runtime_common::messages
> pub trait TransactionEstimation<Weight> {
> 	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight>;
> }
> 
> pub struct X<
> 	AccountId: MaxEncodedLen,
> 	const MaxConfirmationTxWeight: Weight,
> 	const ExtraStorageProofSize: u32,
> 	const TxExtraBytes: u32,
> >(PhantomData<AccountId>);
> 
> impl<AccountId: MaxEncodedLen, const MaxConfirmationTxWeight: Weight, const ExtraStorageProofSize: u32, const TxExtraBytes: u32> TransactionEstimation<Weight> for X<AccountId, MaxConfirmationTxWeight, ExtraStorageProofSize, TxExtraBytes> {
> 	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight> {
> 		let inbound_data_size = InboundLaneData::<AccountId>::encoded_size_hint_u32(1, 1);
> 		MessageTransaction {
> 			dispatch_weight: MaxConfirmationTxWeight,
> 			size: inbound_data_size
> 				.saturating_add(ExtraStorageProofSize)
> 				.saturating_add(TxExtraBytes),
> 		}
> 	}
> }
> 
> // in millau/runtime:
> type MillauTransactionEstimation = bridge_runtime_common::messages::X::<
> 	bp_millau::AccountId,
> 	{ bp_millau::MAX_SINGLE_MESSAGE_DELIVERY_CONFIRMATION_TX_WEIGHT },
> 	{ bp_millau::EXTRA_STORAGE_PROOF_SIZE },
> 	{ bp_millau::TX_EXTRA_BYTES },
> >;
> ```
> 
> And it compiles. TBH weight as a generic argument seems like an overengineering, but right now some weights update is coming, so let's not remove it now.

Thanks ! It works. I must have missed something. I did the change.

File: bin/runtime-common/src/messages.rs [hunk 1]
```diff
@@ -99,12 +99,52 @@ pub struct MessageTransaction<Weight> {
 	pub size: u32,
 }
 
+/// Helper trait for estimating the size and weight of a single message delivery confirmation
+/// transaction.
+pub trait TransactionEstimation<Weight> {
```
Comment (svyatonik) [inline bin/runtime-common/src/messages.rs:104]: One last thing - can you, please, change this name to something like `ConfirmationTransactionEstimation` or something like that? There's another (delivery) transaction here and it isn't "estimated" by this trait.

File: bin/runtime-common/src/messages.rs [hunk 2]
```diff
@@ -99,12 +99,52 @@ pub struct MessageTransaction<Weight> {
...
+	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight>;
```
Comment (svyatonik) [inline bin/runtime-common/src/messages.rs:105]: Could you, please, add some docs for this public method and for the public `BasicTransactionEstimation` struct?

**Review by svyatonik — APPROVED** [review]
Thanks, all is good now!

File: bin/runtime-common/src/messages.rs (hunk 2 above)
Comment (serban300) [reply bin/runtime-common/src/messages.rs:106]: Done

File: bin/runtime-common/src/messages.rs (hunk 1 above)
Comment (serban300) [reply bin/runtime-common/src/messages.rs]: Makes sense. Done.

**Review by svyatonik — APPROVED** [review]

**Pull request merged**